day3.sql

```diff
@@ -0,0 +1,316 @@
+-- FROM  employee e
+-- GROUP BY e.department_id;
+SELECT COALESCE(TO_VARCHAR(e.department_id), '-') as department_id,
+       COUNT(*) as employee_count 
+FROM  employee e
+GROUP BY e.department_id
+
+UNION
+
+SELECT 'Total' as department_id,
+       COUNT(*) as employee_count 
+FROM employee;
+
+
+-- Question 8
+-- Display the jobs held and the employee count.
+-- Hint: every employee is part of at least 1 job Hint: use the previous questions answer Sample
+-- JobsHeld EmpCount 
+-- 1 100 
+-- 2  4
+select cnt,count(cnt) as employee_count from
+(
+(select employee_id,count(employee_id) as cnt from job_history group by employee_id )
+union 
+(select employee_id,count(employee_id) as cnt from employee group by employee_id order by employee_id)
+)
+group by cnt;
+-- SELECT employee_id, COUNT(*) as employee_count
+-- FROM (
+--   SELECT employee_id
+--   FROM job_history
+--   INTERSECT
+--   SELECT employee_id
+--   FROM employee
+-- ) t
+-- GROUP BY employee_id;
+
+-- Question 9
+-- Display average salary by department and country.
+select c.country_name,dept.department_id,avg(emp.salary) as employee_salary
+from employee emp 
+join departments dept
+on emp.department_id = dept.department_id
+join locations loc 
+on dept.location_id = loc.location_id
+join countries c
+on c.country_id = loc.country_id
+group by c.country_name,dept.department_id order by c.country_name; 
+
+-- Question 10
+-- Display manager names and the number of employees reporting to them by countries (each employee works for only one department, and each department belongs to a country)
+select concat(e2.first_name,' ',e2.last_name) as manager_name , count(e1.manager_id) as Employees_Reporting,c.country_name
+from employee e1
+join departments dept
+on e1.department_id = dept.department_id
+join locations loc
+on dept.location_id = loc.location_id
+join countries c 
+on c.country_id = loc.country_id
+join employee e2 
+on e2.employee_id = e1.manager_id
+group by e1.manager_id, c.country_id, manager_name, c.country_name;
+    
+-- select count(employee_id) from employee where manager_id = 101;
+-- select * from 
+-- select * from DEPARTMENTS;
+-- select * from countries ;
+-- select * from locations;
+-- select * from departments;
+-- select * from employee where first_name = 'Neena';
+
+
+-- Question 11
+ -- Group salaries of employees in 4 buckets eg: 0-10000, 10000-20000,.. (Like the previous question) but now group by department and categorize it like below.
+--  DEPT ID 0-10000 10000-20000 
+--  50         2     10
+-- 60          6      5
+-- SELECT department_id,
+--        count(case when salary >= 0 and salary <= 10000 then 1 end) AS "0-10000",
+--        count(case when salary > 10000 and salary <= 20000 then 1 end) AS "10000-20000",
+--        count(case when salary > 20000 and salary <= 30000 then 1 end) AS "20000-30000",
+--        count(case when salary > 30000 then 1 end ) AS "Above 30000"
+-- FROM employee
+-- GROUP BY department_id;
+
```

Comment:
If we want to count the number of rows based on a specific condition  COUNT can be used. In case if we want to calculate the sum of a specific column SUM can be used. If we want to group the data by a specific column and perform an aggregate calculation on each column,  SUM can be used with GROUP BY.